src/dmd/iasmdmd.d

```diff
@@ -2275,6 +2280,11 @@ private void asm_merge_opnds(ref OPND o1, ref OPND o2)
         else
             o1.pregDisp2 = o2.pregDisp2;
     }
+
+    if (o1.bRIP && (o1.pregDisp1 || o2.bRIP))
+        return illegalAddressError("o1.pregDisp1 && RIP");
```

Comment:
No test ?

Reply:
I figured I'd deal with that with the fix for https://issues.dlang.org/show_bug.cgi?id=18373